content/guides/higher-order-functions.adoc

```diff
@@ -0,0 +1,184 @@
+= Higher Order Functions
+Michael Zavarella
+2017-09-15
+:jbake-type: guides
+:toc: macro
+:icons: font
+
+ifdef::env-github,env-browser[:outfilesuffic: .adoc]
+
+toc:[]
+
+== First Class Functions
+
+In functional programming, functions are first class citizens.
+This means functions can be treated as values.
+They can be assigned as values, passed into functions, and returned from functions.
+
+It's common to see function definitions in Clojure using `defn` like `(defn foo ...)`.
+However, this is just syntactic sugar for `(def foo (fn ...))`
+`fn`, and implicitly `defn`, yield a function object as its value that can be passed around like any other Clojure object.
+
+== Higher Order Functions
+
+A higher order function is a function that either:
+
+1. Takes one or more functions as arguments
+2. Returns a function as its result
+
+This will turn out to be a important concept in functional programming in any language.
+
+Higher order functions allow us to _compose_ functions.
+This means we can write small functions and combine them to create larger functions.
+Like putting a bunch of small lego bricks together to build a house.
+
+Let's move away from theory a bit and look at an example.
+
+=== Functions as Arguments
+
+Let's look at two functions
+
+[source, clojure]
+----
+(defn double-+
+    [a b]
+    (* 2 (+ a b)))
+----
+
+[source, clojure]
+----
+(defn double-*
+    [a b]
+    (* 2 (* a b)))
+----
+
+These functions share a common pattern.
+They only differ in name and the function used in the computation of `a` and `b`.
+In general, the pattern looks like this.
+
+[source, clojure]
+----
+(defn double-<f>
+    [a b]
+    (* 2 (f a b)))
+----
+
+We can generalize our `double-` function by passing `f` in as an argument.
+
+[source, clojure]
+----
+(defn double-op
+    [f a b]
+    (* 2 (f a b)))
+----
+
+We can use this to express the concept of doubling the result of an operation rather than having to write functions that perform specific doubled operations individually.
+
+== Anonymous Functions
+
+An anonymous function is a function without a name.
+In Clojure these can be defined in two ways, `fn` and the literal `#(...)`.
```

Comment:
I'd be a little more careful in this section. `fn` does (optionally) take a name, and this is sometimes a useful technique (as the name ends up in function class, which ends up in your stack trace) or to allow for recursive self-calls. 

Maybe the entire section would be better named "Function Literals".